cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java

```diff
@@ -717,25 +711,34 @@ public void setLogLevel(IdeLogLevel logLevel) {
   }
 
   /**
-   * Finds the matching {@link Commandlet} to run, applies {@link CliArguments} to its
-   * {@link Commandlet#getProperties() properties} and will execute it.
+   * Finds the matching {@link Commandlet} to run, applies {@link CliArguments} to its {@link Commandlet#getProperties() properties} and will execute it.
    *
    * @param arguments the {@link CliArgument}.
    * @return the return code of the execution.
    */
   public int run(CliArguments arguments) {
 
+    if (this.ideHome == null) {
+      throw new CliException(getMessageIdeHomeNotFound());
+    }
+
     CliArgument current = arguments.current();
     if (!current.isEnd()) {
       String keyword = current.get();
       Commandlet firstCandidate = this.commandletManager.getCommandletByFirstKeyword(keyword);
       boolean matches;
-      if (firstCandidate != null) {
+      if (firstCandidate == null) {  // no arguments
+        if (current.getArgs().isEmpty()) {
+          info(getMessageIdeHomeFound());
+          return ProcessResult.SUCCESS;
```

Comment:
Better remove the return, otherwise we can't show the help if no arguments were provided.
```suggestion

```

Reply:
I do not get this change.
If the `firstCandidate` was not found, the first argument is not the keyword of a `Commandlet`.
This can happen e.g. if a add an option to a commandlet (see e.g. #279).
In such case the second `for` loop below handles the case to find the proper commandlet.
This however has nothing to do with the fact if `IDE_HOME` was found or not.